### PR TITLE
fix(ingest): hide deprecated path_spec option from config

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/s3/config.py [hunk 1]
```diff
@@ -20,18 +20,14 @@
 
 
 class DataLakeSourceConfig(PlatformSourceConfigBase, EnvBasedSourceConfigBase):
-    path_specs: Optional[List[PathSpec]] = Field(
+    path_specs: List[PathSpec] = Field(
         description="List of PathSpec. See below the details about PathSpec"
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/s3/config.py:24]: Can we replace the `See below` part of the description with a link#anchor to the PathSpec documentation?

File: metadata-ingestion/src/datahub/ingestion/source/s3/config.py [hunk 2]
```diff
@@ -64,23 +60,25 @@ class DataLakeSourceConfig(PlatformSourceConfigBase, EnvBasedSourceConfigBase):
...
+                "path_specs" not in values
+            ), 'cannot have both "path_spec" and "path_specs"'
+            logger.warning("path_spec is deprecated. Please use path_specs instead.")
+            return [values.pop("path_spec")]
+        return v
 
+    @pydantic.root_validator(pre=False)
+    def check_path_specs_and_infer_platform(cls, values: Dict) -> Dict:
         bucket_name: str = ""
+        path_spec: PathSpec
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/s3/config.py:81]: should the code below (lines 93-95 in new) be replaced with?
```python
            if values.get("platform", ""):
                if values["platform"] != platform:
                    raise ValueError("all path_spec should belong to the same platform")
```

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/s3/config.py:81]: yup changed

**Review by rslanka — APPROVED** [review]
LGTM!

**Pull request merged**